pkg/healthv2/provider.go

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package healthv2
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/cilium/cilium/pkg/healthv2/types"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/statedb/index"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+var logger = logging.DefaultLogger.WithField(logfields.LogSubsys, "healthv2")
+
+type HealthProviderParams struct {
+	cell.In
+	Lifecycle cell.Lifecycle
+	DB        *statedb.DB
+}
+
+type HealthProvider struct {
+	db           *statedb.DB
+	stopped      atomic.Bool
+	StatusTable  statedb.RWTable[types.Status]
+	Statuses     statedb.Table[types.Status]
+	PrimaryIndex statedb.Index[types.Status, types.HealthID]
+	LevelIndex   statedb.Index[types.Status, types.Level]
+}
+
+const HealthTable = "health"
```

Comment:
For consistency with definitions elsewhere we should call this `HealthTableName`.